### PR TITLE
Improving the selectors

### PR DESCRIPTION
**Pull request by juan45202**
In the different pages the behavior of the checkout button was repeated so i created a class with this behaivor to optimize the function.

with this change the address and shopping class are not used.

test check
image

### CODE REVIEW

File: cypress/page/general-step.page.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class GeneralStepPage {
+    
+    private proceed: string;
+
+    constructor() {
+        this.proceed = ".cart_navigation .button span";
```
[the remark this answers is not in the document]
Comment (juan45202) [reply cypress/page/general-step.page.ts:6]: I'm trying to use the contains method like this cy.contains("Proceed to checkout").click(); but it doesn't work. it seems to be looking at the top of the page and not the full page. if I use .get and contains it works but it seems redundant cy.get(".cart_navigation").contains("Proceed to checkout").click();

File: cypress/page/payment-step.page.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ class PaymentStepPage {
     constructor() {
         this.bankWire = ".bankwire";
         this.proceed = "#cart_navigation > .button > span";
-        this.elementToCheck = ".cheque-indent > .dark";        
+        this.elementToCheck = ".box .cheque-indent > .dark";
```
[the remark this answers is not in the document]
Comment (juan45202) [reply cypress/page/payment-step.page.ts:10]: Done

File: cypress/page/shipping-step.page.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ class ShippingStepPage {
     private proceed: string
 
     constructor() {
-        this.terms = "#cgv";
+        this.terms = ".order_carrier_content #cgv";
```
[the remark this answers is not in the document]
Comment (juan45202) [reply cypress/page/shipping-step.page.ts:7]: Done

File: cypress/e2e/buy-shirt.cy.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import {MenuContentPage, AddressStepPage, LoginPage, PaymentStepPage,
-  ProductsListPage, ShippingStepPage, ShoppingCartPage} from "../page/index";
+  ProductsListPage, ShippingStepPage, ShoppingCartPage, GeneralStepPage} from "../page/index";
 
 const menuContentPage = new MenuContentPage();
 const addressStepPage = new AddressStepPage();
```
Comment (manuelq12) [inline cypress/e2e/buy-shirt.cy.ts:5]: I'm seeing a warning that this instance is not being used anywhere, please check

File: cypress/e2e/buy-shirt.cy.ts [hunk 5]
```diff
@@ -8,6 +8,7 @@ const paymentStepPage = new PaymentStepPage();
 const productsListPage = new ProductsListPage();
 const shippingStepPage = new ShippingStepPage();
 const shoppingCartPage = new ShoppingCartPage();
```
Comment (manuelq12) [inline cypress/e2e/buy-shirt.cy.ts:10]: I'm seeing a warning that this instance is not being used anywhere, please check

File: cypress/e2e/buy-shirt.cy.ts (hunk 4 above)
Comment (juan45202) [reply cypress/e2e/buy-shirt.cy.ts:5]: Done

File: cypress/e2e/buy-shirt.cy.ts (hunk 5 above)
Comment (juan45202) [reply cypress/e2e/buy-shirt.cy.ts:10]: Done

**Review by manuelq12 — APPROVED** [review]
I'll approve, I think the pages names are not met but for you to continue

**Pull request merged**